substrate/frame/identity/src/lib.rs

```diff
@@ -886,6 +966,181 @@ pub mod pallet {
 			});
 			Ok(())
 		}
+
+		/// Add an `AccountId` with permission to grant usernames with a given `suffix` appended.
+		/// The authority can grant up to `allocation` usernames. To top up their allocation, they
+		/// should just issue (or request via governance) a new `add_username_authority` call.
+		#[pallet::call_index(15)]
+		#[pallet::weight(T::WeightInfo::add_username_authority())]
+		pub fn add_username_authority(
+			origin: OriginFor<T>,
+			authority: AccountIdLookupOf<T>,
+			suffix: Vec<u8>,
+			allocation: u32,
```

Comment:
Should there be a maximum allocation allowed per authority?

Reply:
The allocation needs to come from governance anyway, so IMO that's just something to cause trouble (i.e. failing referenda).